cogs/utils/arg_converter.py

```diff
@@ -0,0 +1,150 @@
+        dictionary is used. This could lead to unexpected results:
+        for example, it doesn't matter if this is to catch two discord users,
+        but if there was a converter meant to catch any string, then a
+        converter to catch an uppercase string, then the arguments
+        ["HI", "hello"] will raise an exception, since the "any" string
+        converter is the one used with "HI", and then the only remaining
+        converter is the uppercase one, and so "hello" is not converted.
+
+        Returns a dictionary with the variables as keys, and their values.
+        """
+        # initialize the converted dict with None values
+        converted_arguments_dict = dict.fromkeys(self.converters_dict)
+        # get a copy of the list of variables that are required
+        # if there are still values in this at the end, an error will be raised
+        remaining_req_vars = self.req_vars.copy()
+        for arg in arguments:
+            # keep track if the argument is converted. If it is not after
+            # trying every converter, an error will be raised
+            arg_converted = False
+            for key in self.converters_dict:
+                try:
+                    # if the value for that key
+                    # is still none in the converted dict
+                    if not converted_arguments_dict[key]:
+                        # try converting the argument
+                        converted_arguments_dict[
+                            key] = await self.converters_dict[key][0].convert(
+                                ctx, arg)
+                        # if we did, delete the key from the converters dict
+                        # as there can only be one value per key
+                        del self.converters_dict[key]
+                        # if this key was required, remove it from the list
+                        # of remaining required keys
+                        if key in remaining_req_vars:
+                            remaining_req_vars.remove(key)
+                        arg_converted = True
+                        break
+                except commands.BadArgument:
+                    # if this converted didn't work, we try the next
+                    pass
+            if not arg_converted:
+                raise commands.BadArgument(
+                    f"Invalid input: Argument \"{arg}\" could not be converted"
+                )
+        # if there are still required variables
+        if remaining_req_vars:
+            raise commands.BadArgument(
+                f"Invalid input: Missing required argument"
+                f"{'s' if len(remaining_req_vars)>1 else ''} "
+                f"{str(remaining_req_vars)[1:-1]}")
+        # for values that are still None, replace them by their default value
+        # if applicable
+        for key, value in converted_arguments_dict.items():
+            if not value:
```

Comment:
use `if value is None` instead, since `value` could be `False`

Reply:
That was a good catch, however this part is now outdated in c6111c6